Yoast/javascript
# P2 786 only render ok when block is valid

## Summary


This PR can be summarized in the following changelog entry:

* [@yoast/schema-blocks] Shows the ok only when the added block suggestion hasn't any validation issues.

## Relevant technical choices:

* removed unneeded transform in jest config.
* prevent error in Instruction base class.
* Added required option to BaseInstruction to reduce code duplication.
* Moved default validation logic to a function; not all blocks need validation at all. Now, validation is UNKNOWN by default, or blocks can (re)use the default validation function.
* Moved some definitions to their own files.
* Cleaned up `/src/core/blocks` imports / exports.
* Cleaned up `src/core/validation` exports.
* De-duplicated an error code.
* Removed duplicate code.
* extracted logic from `BlockSuggestionsPresenter.tsx` so the presenter only presents.
* used a store subscriber in `PureBlockSuggestionsPresenter` to update the suggestions as they change.
* added new selectors: `getValidationResultForClientId`,`getInnerblockValidations`
* added some defensive input checks 
* new helpder function `recursivelyfind` to navigate the tree of BlockValidationResults using a predicate.
* fine-tuned validation logic of some instructions to match new (default-) validation logic
* unit tests.

## Test instructions

This PR can be tested by following these steps:

* Add a job block to a new post
* See that most of the items do not have a green checkmark in the sidebar. The analysis conclusion is still RED.
* complete the template for all required blocks shown in the sidebar. Check marks should appear instantly when a non-whitespace character is added.
* When all required blocks are completed, verify that the `good job` congratulation is shown.

## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  *

## UI changes

## Quality assurance


Fixes #
